x-pack/elastic-agent/pkg/agent/application/upgrade/step_download.go

```diff
@@ -20,7 +22,17 @@ import (
 	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/release"
 )
 
-func (u *Upgrader) downloadArtifact(ctx context.Context, version, sourceURI string) (string, error) {
+func (u *Upgrader) downloadArtifact(ctx context.Context, version, sourceURI string) (_ string, err error) {
+	// TODO: span type?
+	span, ctx := apm.StartSpan(ctx, "Upgrader.downloadArtifact()", "custom")
+	defer func() {
+		span.End()
```

Comment:
If you call this before capturing the error, the error's timestamp may be after the span ends. I'd reverse the order.